mbed_greentea/mbed_greentea_cli.py

```diff
@@ -726,140 +645,116 @@ def main_cli(opts, args, gt_instance_uuid=None):
-                    disk = mut['mount_point']
-                    port = mut['serial_port']
-                    micro = mut['platform_name']
-                    program_cycle_s = mut_info_map[platfrom_name]['properties']['program_cycle_s']
-                    copy_method = opts.copy_method if opts.copy_method else 'shell'
-                    enum_host_tests_path = get_local_host_tests_dir(opts.enum_host_tests)
-
-                    yotta_config = YottaConfig()
-                    yotta_config.init(yotta_target_name)
-
-                    yotta_config_baudrate = yotta_config.get_baudrate()
-
-                    # We will force configuration specific baudrate
-                    if port:
-                        port = "%s:%d"% (port, yotta_config_baudrate)
-
-                    test_platforms_match += 1
-                    host_test_result = run_host_test(opts.run_app,
-                                                     disk,
-                                                     port,
-                                                     yotta_target_name,
-                                                     mut['target_id'],
-                                                     micro=micro,
-                                                     copy_method=copy_method,
-                                                     program_cycle_s=program_cycle_s,
-                                                     digest_source=opts.digest_source,
-                                                     json_test_cfg=opts.json_test_configuration,
-                                                     run_app=opts.run_app,
-                                                     enum_host_tests_path=enum_host_tests_path,
-                                                     verbose=True)
-
-                    single_test_result, single_test_output, single_testduration, single_timeout, result_test_cases, test_cases_summary = host_test_result
-                    status = TEST_RESULTS.index(single_test_result) if single_test_result in TEST_RESULTS else -1
-                    if single_test_result != TEST_RESULT_OK:
-                        test_exec_retcode += 1
-                    continue
-
-                # Regression test mode:
-                # Building sources for given target and perform normal testing
-
-                yotta_result, yotta_ret = True, 0   # Skip build and assume 'yotta build' was successful
-                if opts.skip_yotta_build:
-                    gt_logger.gt_log("skipping calling yotta (specified with --skip-build option)")
+            if mbed_dev['platform_name'] == platform_name:
+                # We will force configuration specific baudrate
+                if mbed_dev['serial_port']:
+                    mbed_dev['serial_port'] = "%s:%d" % (mbed_dev['serial_port'], baudrate)
```

Comment:
```if mbed_dev['platform_name'] == platform_name:``` or ```if mbed_dev['serial_port']:```
First one is to find a suitable platform for testing this build. It wouldn't do ```muts_to_test.append``` if it fails.
Second, it should not be None. Hence removed that check. But as caution it would be good to add validation on mbedls output.